seam/parse_options.py

```diff
@@ -0,0 +1,89 @@
+import os
+from typing import Optional
+
+DEFAULT_ENDPOINT = "https://connect.getseam.com"
+
+
+def parse_options(
+    api_key: Optional[str] = None,
+    personal_access_token: Optional[str] = None,
+    workspace_id: Optional[str] = None,
+    endpoint: Optional[str] = None,
+):
+    if personal_access_token is None:
+        api_key = api_key or os.getenv("SEAM_API_KEY")
+
+    from seam.auth import get_auth_headers
```

Comment:
@razor-x there was an `ImportError` caused by a circular import between `parse_options.py` and `auth.py`, where each file was importing from the other. I fixed it using a lazy import as shown above. Does it look good to you?

Alternatively, we could move the `get_auth_headers` function from `auth.py` to `parse_options.py` to eliminate the need for `parse_options.py` to import from `auth.py`. Another option is to introduce a third module to import stuff from.

Reply:
Let's avoid the lazy import, it's easy to miss. We can move the `is_seam_http_options_with_personal_access_token` functions and `SeamHttpInvalidOptionsError` stuff to `options.py`. Let `parse_options.py` only export the `parse_options` function